Composer/packages/client/src/App.tsx

```diff
@@ -32,6 +34,7 @@ export const App: React.FC = () => {
     checkNodeVersion();
     fetchExtensions();
     fetchFeatureFlags();
+    performAppCleanupOnQuit();
```

Comment:
Shouldn't this be invoke when electron sends the `'cleanup'` event? How is this supposed to be triggered when the user is running it in the browser?

```js
const { ipcRenderer } = window;

useEffect(() => {
  ipcRenderer?.on('cleanup', (_event) => {
      performAppCleanupOnQuit();
  });
}, []);
```

Reply:
It is essentially a no-op for the browser.

My understanding is that since the server (Node process) that runs the app in the browser is responsible for spawning the bot processes, the processes should shut down when the user stops the server. The user exiting the browser tab shouldn't necessarily stop the server.

Reply:
Oh I see now. I was moving files around and I forgot to add the ipc code that I previously had. Yes, it should have a listener like that.